apis/externalsecrets/v1alpha1/secretstore_alibaba_types.go

```diff
@@ -33,9 +44,7 @@ type AlibabaAuthSecretRef struct {
 
 // AlibabaProvider configures a store to sync secrets using the Alibaba Secret Manager provider.
 type AlibabaProvider struct {
-	Auth *AlibabaAuth `json:"auth"`
-	// +optional
-	Endpoint string `json:"endpoint"`
```

Comment:
no, because AlibabaAuth was
```
type AlibabaAuth struct {
	SecretRef AlibabaAuthSecretRef `json:"secretRef"`
}
```

ake you had to pass access key id and secret

for RRSA I changed it to
```
type AlibabaAuth struct {
	// +optional
	SecretRef *AlibabaAuthSecretRef `json:"secretRef,omitempty"`
	// +optional
	RRSAAuth *AlibabaRRSAAuth `json:"rrsa,omitempty"`
}
```

as when using RRSA you don't pass any of the previous auth fields. and vice versa

Reply:
The `Endpoint` field gets removed - If users have a resource with an endpoint specified this resource gets rejected.